src/dev-app/focus-trap/focus-trap-demo.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {FocusTrap, FocusTrapFactory} from '@angular/cdk/a11y';
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  ViewChild,
+  ViewEncapsulation} from '@angular/core';
+import {MatDialog} from '@angular/material/dialog';
+
+@Component({
+  selector: 'shadow-dom-demo',
+  template: '<ng-content></ng-content>',
+  host:     {'class': 'demo-focus-trap-shadow-root'},
+  encapsulation: ViewEncapsulation.ShadowDom
+})
+export class FocusTrapShadowDOMDemo {}
```

Comment:
```suggestion
export class FocusTrapShadowDomDemo {}
```
(PascalCase even for acronyms)